api/models/linkPreview/findOrCreateByUrl.js

```diff
@@ -0,0 +1,27 @@
+export default function findOrCreateByUrl (url) {
+  return LinkPreview.find(url).then(preview => {
+    if (!preview) {
+      return LinkPreview.queue(url)
+      .then(() => {
+        return {status: 'queued'}
+      })
+    }
+
+    if (!preview.get('done')) {
+      return {status: 'loading'}
+    }
+
+    return Object.assign(
+      preview.pick(
+        'id',
+        'url',
+        'image_url',
+        'title',
+        'description',
+        'image_width',
+        'image_height'
+      ),
+      {status: 'loaded'}
```

Comment:
Since this value is never used by the front-end, all you really need to do here is return the `preview` instance.